src/compartment.c

```diff
@@ -231,6 +249,15 @@ comp_map(struct Compartment *to_map)
                 &to_map->libs[i]->rela_maps[j].target_func_address,
                 sizeof(void *));
         }
+        // Map .tbss sections
+        if (to_map->libs[i]->tls_data_size != (size_t) -1)
+        {
+            assert(to_map->libs[i]->tls_sec_addr);
+            // TODO why 4?
```

Comment:
Indeed... Why 4?!

Reply:
I, uh, don't know. It seems to work with `sizeof(void*)`, so I changed the question now to "Why `sizeof(void*)`. This might be related to why `printf` doesn't work, but can't say without some in-depth debugging.

Reply:
Alright, managed to uncover and fix an issue with subsequent library TLS regions (as I hinted) in https://github.com/capablevms/CHERI-ELF-comp/pull/28/commits/f47b3c959e7c590b3a25747bf7009422c0a91372. This was also part of that fix.[](url)